x-pack/plugins/cloud_security_posture/server/tasks/findings_stats_task.ts

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { transformError } from '@kbn/securitysolution-es-utils';
+import {
+  RunContext,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import { SearchRequest } from '@kbn/data-plugin/common';
+import { ElasticsearchClient } from '@kbn/core/server';
+import type { Logger } from '@kbn/core/server';
+import {
+  AggregatedFindingsByCluster,
+  ScoreBucket,
+  FindingsStatsTaskResult,
+  TaskHealthStatus,
+} from './types';
+import {
+  BENCHMARK_SCORE_INDEX_DEFAULT_NS,
+  LATEST_FINDINGS_INDEX_DEFAULT_NS,
+} from '../../common/constants';
+import { scheduleTaskSafe, removeTaskSafe } from '../lib/task_manager_util';
+import { CspServerPluginStartServices } from '../types';
+
+const CSPM_FINDINGS_STATS_TASK_ID = 'cloud_security_posture-findings_stats';
```

Comment:
It would be better if the ID and the TYPE won't have the same value, so we could distinguish between the two